### DISCUSSION TITLE
USWDS 3.0 Beta Program

### DISCUSSION THREAD

**Discussion by thisisdano** [uswds/uswds#4365] [General]
Next week we'll be kicking off the USWDS 3.0 Beta program. ✨ For a bit of background about what we're trying to do with USWDS 3.0, see the [recap here in discussions](https://github.com/uswds/uswds/discussions/4355) and the slide deck from [our September 2021 monthly call](https://digital.gov/event/2021/09/16/uswds-monthly-call-september-2021/). 

We're going to need some help testing what we build. Getting feedback from real people, products, and projects will be critical to the success of this release. So we'll be publishing beta releases of our 3.0 development branch via npm at least monthly until the official USWDS 3.0 release early next year. And we need you to put these releases through the paces. (Or just install them and make sure we aren't breaking anything!) 

Our goal is to get to a USWDS 3.0 release that's as simple as possible to install and upgrade. As we publish each new release, there may be different parts of the upgrade and installation process we'll ask you to assess. The more feedback we can get from you now, the better we can make the process for everyone later.

So if you have the time and inclination, please speak up here and let us know if you'd like to be included in the program. We'll add you to a special USWDS GitHub team, and we'll tag the team on any beta-related discussions or issues moving forward. 

Thank you, we appreciate your help! 🚀

**Comment by rfultz** [comment]
Speaking up

I'm interested in helping

**Comment by alex-hsa** [comment]
Please add me to your list of beta testers.
Thank you, Alex

**Comment by SCourtsMember** [comment]
The Admin Office/U.S. Courts team uses the USWDS on a handful of it's sites and planned to continue using the system on all of our upcoming Drupal 9 websites. Our team is a bit concerned that USWDS will no longer be covering IE 11 in version 3.0. We still have a significant amount of gov users who will be using IE11. I'm curious if there is a process in place to phase out coverage by keeping the current version available for use posted online longer or how other gov sites are handling this change. I'm also open to testing.

**Reply by mejiaj** [reply]
We're not looking to rip out all of the IE11 functionality on USWDS 3. Not right away at least. The plan is just to not support it moving forward.

**Reply by SCourtsMember** [reply]
Do you have any details on what the phasing out of IE11 functionality on USWDS 3 will look like? Timeline?

**Comment by mejiaj** [comment]
Thanks to everyone for their interest in helping us test. An even bigger thanks for your patience as we were getting stable release [2.12.2 ](https://github.com/uswds/uswds/releases/tag/v2.12.2) out the door.

We'll be syncing the latest 2.12.2 changes into this beta branch and release shortly after.

**Comment by dara-icf** [comment]
I would like to participate in the beta.

**Comment by thescientist13** [comment]
Hope it's not too late, but a fellow OSS maintainer (https://github.com/uswds/uswds/discussions/4312#discussioncomment-1599697) and would like to try and help where I can.  Not an "active" / power user but as a citizen, I very much find value in your work and would like to be find ways to get involved.

**Comment by mbucknell** [comment]
I know I am late to the game but I would like to try and help. As part of my work, I have two projects using USWDS and a couple others that are in maintenance. Our applications use a combination of static pages and dynamic javascript (D3 in one case, Vue in the other).

**Reply by mejiaj** [reply]
Not too late @mbucknell ! We've released the first beta of v3 and started a discussion here https://github.com/uswds/uswds/discussions/4386

We'll be having follow up discussions per every release. If you have any issues with the first release please post them there. Otherwise we'll continue to leave this discussion open to get more potential testers.

**Comment by EricRohlfs** [comment]
I'm interested. I'm very interested in ES Imports working directly in the browser (no tooling like webpack, rollup, or other). I was able to get them working in the 2.latest version using rollup with a few plug-ins.

Also, has there been any discussion on supporting customElements in addition to classes. I tried making one for the modal over the weekend and didn't seem overly challenging. Well, customElements for things that make sense as custom elements.

`

**Reply by mejiaj** [reply]
Sorry for missing this @EricRohlfs!

>I'm interested. I'm very interested in ES Imports working directly in the browser (no tooling like webpack, rollup, or other). I was able to get them working in the 2.latest version using rollup with a few plug-ins.

We have this issue for ES Modules https://github.com/uswds/uswds/issues/4515. While we'd love to get there soon, it's a big effort alone and a breaking change so we have to proceed carefully. If you have any advice/examples that would be much appreciated.

>Also, has there been any discussion on supporting customElements in addition to classes. I tried making one for the modal over the weekend and didn't seem overly challenging. Well, customElements for things that make sense as custom elements.

This would be the same as web components right? That's another future goal for select components.

**Comment by ryparker** [comment]
The USWDS 3.0 beta has ended and 3.0 has been released, so I'm closing this old recruitment post. For help with a current version, please use [Q&A](https://github.com/uswds/uswds/discussions/categories/q-a).
